 use p->p_vaddr as loadaddr in yaboot, not the ELF entry point

Index: yaboot/second/yaboot.c
===================================================================
--- yaboot/second/yaboot.c	(revision 301)
+++ yaboot/second/yaboot.c	(revision 302)
@@ -1151,7 +1151,7 @@ load_elf32(struct boot_file_t *file, loa
           loadaddr = KERNELADDR;
      } else {
           flat_vmlinux = 0;
-          loadaddr = e->e_entry;
+          loadaddr = loadinfo->load_loc;
      }
 
      /* On some systems, loadaddr may already be claimed, so try some
